packages/flutter/lib/src/services/text_input.dart

```diff
@@ -846,6 +847,127 @@ mixin TextSelectionDelegate {
 
   /// Whether select all is enabled, must not be null.
   bool get selectAllEnabled => true;
+
+  /// Cut current [selection] to Clipboard.
+  ///
+  /// If [cause] is [SelectionChangedCause.toolBar], the position of
+  /// [bringIntoView] to selection will be called and hide toolBar.
+  ///
+  /// {@macro flutter.rendering.RenderEditable.cause}
+  void cutSelection(SelectionChangedCause cause) {
+    final TextSelection selection = textEditingValue.selection;
+    final String text = textEditingValue.text;
+    Clipboard.setData(ClipboardData(
```

Comment:
What if the selection is collapsed?  I see we were checking for that before in RenderEditable's cutSelection.

Reply:
The code to check that selection is collapsed is still available in RenderEditable's cutSelection.